scripts/add_electricity.py

```diff
@@ -247,6 +266,14 @@ def attach_load(n, regions, load, admin_shapes, countries, scale):
         gpd.read_file(regions).set_index("name").reindex(substation_lv_i)
     ).dropna(
         axis="rows")  # TODO: check if dropna required here. NaN shapes exist?
+
+    weather_year = snakemake.config["load_options"]["weather_year"]
```

Comment:
In the current design, you have to move the variable assignment out of the function. Assigning variables in the functions is not the recommended way. I will mark somewhere at the bottom of the script where to assign the variables.

Reply:
Here at the bottom of the add_electricity script is the execution of the functions. There you can also see that the  `attach_load()` function is executed. 
https://github.com/pypsa-meets-africa/pypsa-africa/blob/2cd796463004d8aeb8751cf21d8fe466a85936cb/scripts/add_electricity.py#L692-L715

So what you need to do:
1. At the execution part add your new function which has some extra variables `attach_load(n, regions, weather_year, prediction_year, region_load, ssp, admin_shapes, countries, scale)`
2. Some lines before you have to assign the variables which you need for the function. For instance:
```
prediction_year = snakemake.config["load_options"]["prediction_year"]
region_load = snakemake.config["load_options"]["region_load"]
....
```